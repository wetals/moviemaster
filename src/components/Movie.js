import React from "react";
import Genre from "./Genre";

const Movie = props => {
  const { title, genre_ids, poster_path } = props.movie;

  const { base_url, poster_sizes } = props.images;
  const image_path = `${base_url}${poster_sizes[3]}${poster_path}`;

  return (
    <article className="item">
      <div className="item__image">
        <img src={image_path} alt={title} />
      </div>
      <div className="item__metadata">
        <div className="item__title">{title}</div>
        <Genre genre_ids={genre_ids} genresList={props.genresList} />
      </div>
    </article>
  );
};

export default Movie;
