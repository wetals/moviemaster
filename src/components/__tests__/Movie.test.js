import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import Movie from "../Movie";

const testData = {
  movie: {
    title: "foo",
    genre_ids: [],
    poster_path: "1.jpg"
  },
  images: {
    base_url: "url",
    poster_sizes: ["size-1", "size-2", "size-3", "size-4", "size-5"]
  },
  genresList: {}
};

describe("Movie component", () => {
  it("renders properly", () => {
    const { movie, images } = testData;
    const image_path = `${images.base_url}${images.poster_sizes[3]}${
      movie.poster_path
    }`;
    const wrapper = shallow(<Movie {...testData} />);
    const element = wrapper.find("article");
    expect(element.find("img").prop("src")).toEqual(image_path);
    expect(element.find(".item__title").text()).toBe("foo");
  });

  it("renders snapshot properly", () => {
    const tree = renderer.create(<Movie {...testData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
