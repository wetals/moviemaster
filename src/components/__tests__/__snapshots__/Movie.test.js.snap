// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component renders snapshot properly 1`] = `
<article
  className="item"
>
  <div
    className="item__image"
  >
    <img
      alt="foo"
      src="urlsize-41.jpg"
    />
  </div>
  <div
    className="item__metadata"
  >
    <div
      className="item__title"
    >
      foo
    </div>
    <div
      className="genre__tags"
    />
  </div>
</article>
`;
